src/components/fileviewer.js

```diff
@@ -49,12 +54,43 @@ export default class FileViewerContainer extends Component {
       "format": "list"
     })
     .then(data => {
-      this.setState({coverage: data});
       // TODO remove these log lines
       console.log(data);
+      this.setState({coverage: data});
+      this.parseTestsCoverage(data.data);
     });
   }
 
+  /* Parse data returns from ActiveData */
+  parseTestsCoverage(data) {
+    console.log(data.length);
+    var stat = []
+
+    for (var i = 0; i < data.length; i++) {
+      var coveredLines = data[i].source.file.covered;
+      for (var j = 0; j < coveredLines.length; j++) {
+        var line = coveredLines[j];
+        if (!stat[line]) {
+          stat[line] = [];
+        }
+        stat[line].push(i);
```

Comment:
Instead of `stat[line].push(i)`  use `stat[line].push(d)` then

    const testItems = testList.map((testNum) => {
      const test = coverage.data[testNum];
      const testName = test.run.name;
      return <li key={testName}>{testName}</li>
    });

will read

    const testItems = testList.map((test) => {
      return <li key={test.run.name}>{test.run.name}</li>
    });

which then suggest we inline this on-line function